modules/http/http.c

```diff
@@ -153,6 +153,23 @@ _set_curl_opt(HTTPDestinationDriver *self, LogMessage *msg, struct curl_slist *c
   if (self->user_agent)
     curl_easy_setopt(self->curl, CURLOPT_USERAGENT, self->user_agent);
 
+  if (self->ca_dir)
+    curl_easy_setopt(self->curl, CURLOPT_CAPATH, self->ca_dir);
+
+  if (self->ca_file)
+    curl_easy_setopt(self->curl, CURLOPT_CAINFO, self->ca_file);
+
+  if (self->cert_file)
+    curl_easy_setopt(self->curl, CURLOPT_SSLCERT, self->cert_file);
+
+  if (self->key_file)
+    curl_easy_setopt(self->curl, CURLOPT_SSLKEY, self->key_file);
+
+  if (self->ciphers)
+    curl_easy_setopt(self->curl, CURLOPT_SSL_CIPHER_LIST, self->ciphers);
+
```

Comment:
By default, libcurl will use the default cipher suite compiled into the crypto library.  For OpenSSL this is normally "ALL:!EXPORT:!LOW:!aNULL:!eNULL:!SSLv2".  The compiled in default value can also be specified as "DEFAULT" for OpenSSL and GnuTLS.  The actual values allowed depends on the underlying crypto libs.  OpenSSL and GnuTLS use the same syntax but the syntax accepted by NSS is completely different.  So hard-coding a default value can be problematic since we don't know what crypto libs libcurl was built with.

Adding the ability to specify the SSL/TLS version is a good idea.  By default, libcurl will use TLS 1.x (not SSL).